benchmarks/src/search_n.cpp

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <benchmark/benchmark.h>
+#include <cstddef>
+#include <cstdint>
+#include <limits>
+#include <vector>
+
+#include "skewed_allocator.hpp"
+
+using namespace std;
+
+// NB: This particular algorthm has std and ranges implementations with different perf charactterstitics!
+
+enum class AlgType { Std, Rng };
+
+template <class T, AlgType Alg>
+void bm(benchmark::State& state) {
+    const auto size = static_cast<size_t>(state.range(0));
+
+    constexpr size_t count = 1;

```

Comment:
Nitpick: As `count` is the name of an algorithm, I would prefer to avoid shadowing it here.

(`size` is also a Standard name, but I've stopped complaining about it)